wpilibc/src/main/native/cpp/Tracer.cpp

```diff
@@ -4,14 +4,226 @@
 
 #include "frc/Tracer.h"
 
+#include <memory>
+#include <string>
+#include <thread>
+#include <utility>
+#include <vector>
+
 #include <fmt/format.h>
+#include <networktables/DoubleTopic.h>
+#include <networktables/NetworkTable.h>
+#include <networktables/NetworkTableInstance.h>
+#include <units/time.h>
 #include <wpi/SmallString.h>
+#include <wpi/StringMap.h>
+#include <wpi/deprecated.h>
 #include <wpi/raw_ostream.h>
 
 #include "frc/Errors.h"
+#include "frc/RobotController.h"
 
 using namespace frc;
 
+std::atomic<bool> singleThreadedMode = false;
+std::atomic<bool> anyTracesStarted = false;
+
+class TracerState {
+ public:
+  TracerState() {
+    if (singleThreadedMode && anyTracesStarted) {
+      FRC_ReportWarning("Cannot start a new trace in single-threaded mode");
+      m_disabled = true;
```

Comment:
same here- singleThreadedMode *can't* be true here.

Reply:
Why can't it be true? this constructer is only called in a `thread_local` while `singleThreadedMode` is not thread local. If the main thread sets `singleThreadedMode` to true, makes a trace then starts a new thread will this not report a warning?

Reply:
Sorry, I wasn't clear. It couldn't be true previously because `SetSingleThreadedMode` was locked out by `anyTracesStarted`.